### PR TITLE
tests: replace refresh_metrics() with trigger_processing in statsd

### PR DESCRIPTION
**Pull request by jd**
The current tests can fail just because the timeout it not set to None. The
base method trigger_processing is meant to be used just for that!

### CODE REVIEW

**Comment by jd** [comment]
Pull-request updated, HEAD is now cda6ccfbd85767a3a743e165d9500730d949c343

File: gnocchi/tests/functional/gabbits/aggregates-with-resources.yaml [hunk 1]
```diff
@@ -127,7 +127,7 @@ tests:
         search: "user_id = 'A50F549C-1F1C-4888-A71A-2C5473CCCEC1'"
         operations: "(aggregate mean (metric cpu.util mean))"
       poll:
-        count: 10
+        count: 60
```
Comment (chungg) [inline gnocchi/tests/functional/gabbits/aggregates-with-resources.yaml]: it seems like this is just hiding potentially another issue. pifpaf is polling for work every second so even 10s should be more than enough.

Comment (jd) [reply gnocchi/tests/functional/gabbits/aggregates-with-resources.yaml]: Yeah I was just trying to see if that was the source of the problem, but clearly it's not. Not sure what it is :(

**Comment by jd** [comment]
Pull-request updated, HEAD is now fb37011f933c0bee9b676569325fe91a0f358579

**Comment by jd** [comment]
@chungg yup, done

**Pull request merged**